src/shared/nested-dropdown-list/list-item.tsx

```diff
@@ -0,0 +1,209 @@
+
+                    return found;
+                } else {
+                    const found = defaultExpandKeys.includes(item.key);
+                    return found;
+                }
+        }
+    }
+
+    const checkListItemSelected = (item: V1 | V2 | V3): boolean => {
+        return !!find(selectedItems, (arrItem) => {
+            return isEqual(arrItem, item);
+        });
+    };
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const renderTruncatedText = (
+        item: FItemOption<V1, V2, V3>
+    ): JSX.Element => {
+        const displayText = listExtractor
+            ? listExtractor(item)
+            : item.toString();
+
+        return (
+            <TruncateContainer data-testid="truncate-middle-container">
+                <TruncateFirstLine>{displayText}</TruncateFirstLine>
+                <TruncateSecondLine>{displayText}</TruncateSecondLine>
+            </TruncateContainer>
+        );
+    };
+
+    const hasExceededContainer = (item: FItemOption<V1, V2, V3>) => {
+        const displayText = listExtractor
+            ? listExtractor(item)
+            : item.toString();
+
+        let widthOfElement = 0;
+        if (labelRef && labelRef.current) {
+            widthOfElement = labelRef.current.getBoundingClientRect().width;
+        }
+
+        return StringHelper.shouldTruncateToTwoLines(
+            displayText,
+            widthOfElement
+        );
+    };
+
+    const renderListItem = () => {
+        if (!expand) return;
+
+        const nextSubItems = item.subItems.values();
+
+        return (
+            <ul data-testid="sub-category-list">
+                <List>
+                    <ul>
+                        {[...nextSubItems].map((item) => (
+                            <ListItem
+                                key={item.key}
+                                parentKeys={[...parentKeys, item.key]}
+                                item={item}
+                                mode={mode}
+                                listExtractor={listExtractor}
+                                selectedItems={selectedItems}
+                                selectedKeys={selectedKeys}
+                                defaultExpandKeys={defaultExpandKeys}
+                                visible={visible}
+                                onBlur={onBlur}
+                                onExpand={onExpand}
+                                onSelect={onSelect}
+                            />
+                        ))}
+                    </ul>
+                </List>
+            </ul>
+        );
+    };
+
+    if (!item.subItems) {
+        const hasExceeded = hasExceededContainer(item);
+
+        /**
+         * force re-render dropdown height
+         * buggy happened if flat option in tier 1
+         */
+        if (!hasExceeded) onExpand();
```

Comment:
can you elaborate on what the issue is?

generally it's dangerous to have side effects in the render function. can try putting this logic in a `useEffect`

Reply:
this was removed in subsequent PR. 

I try triggered re-count the content height.